x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -0,0 +1,531 @@
+import os from 'os';
+import { renderSummaryTable } from './print_run';
+import type { SecuritySolutionDescribeBlockFtrConfig } from './utils';
+import { isSkipped, parseTestFileConfig } from './utils';
+
+interface ProductType {
+  product_line: string;
+  product_tier: string;
+}
+
+interface CreateEnvironmentRequestBody {
+  name: string;
+  region_id: string;
+  product_types: ProductType[];
+}
+
+interface Environment {
+  name: string;
+  id: string;
+  region: string;
+  es_url: string;
+  kb_url: string;
+  product: string;
+}
+
+interface Credentials {
+  username: string;
+  password: string;
+}
+
+const DEFAULT_REGION = 'aws-eu-west-1';
+let log: ToolingLog;
+
+/**
+ * Retrieve test files using a glob pattern.
+ * If process.env.RUN_ALL_TESTS is true, returns all matching files, otherwise, return files that should be run by this job based on process.env.BUILDKITE_PARALLEL_JOB_COUNT and process.env.BUILDKITE_PARALLEL_JOB
+ */
+const retrieveIntegrations = (integrationsPaths: string[]) => {
+  const nonSkippedSpecs = integrationsPaths.filter((filePath) => !isSkipped(filePath));
+
+  if (process.env.RUN_ALL_TESTS === 'true') {
+    return nonSkippedSpecs;
+  } else {
+    // The number of instances of this job were created
+    const chunksTotal: number = process.env.BUILDKITE_PARALLEL_JOB_COUNT
+      ? parseInt(process.env.BUILDKITE_PARALLEL_JOB_COUNT, 10)
+      : 1;
+    // An index which uniquely identifies this instance of the job
+    const chunkIndex: number = process.env.BUILDKITE_PARALLEL_JOB
+      ? parseInt(process.env.BUILDKITE_PARALLEL_JOB, 10)
+      : 0;
+
+    const nonSkippedSpecsForChunk: string[] = [];
+
+    for (let i = chunkIndex; i < nonSkippedSpecs.length; i += chunksTotal) {
+      nonSkippedSpecsForChunk.push(nonSkippedSpecs[i]);
+    }
+
+    return nonSkippedSpecsForChunk;
+  }
+};
+
+const encode = (str: string): string => Buffer.from(str, 'binary').toString('base64');
+
+const getApiKeyFromElasticCloudJsonFile = () => {
+  const userHomeDir = os.homedir();
+  try {
+    const jsonString = fs.readFileSync(path.join(userHomeDir, '.elastic/cloud.json'), 'utf-8');
+    const jsonData = JSON.parse(jsonString);
+    return jsonData.api_key.qa;
+  } catch (e) {
+    log.info('API KEY could not be found in ');
+    return null;
+  }
+};
+
+// Poller function that is polling every 20s, forever until function is resolved.
+async function poll<T>(
+  fn: () => Promise<T>,
+  retries: number = 200,
+  interval: number = 20000
+): Promise<T> {
+  return Promise.resolve()
+    .then(fn)
+    .catch(async function retry(err: Error): Promise<T> {
+      let remainingRetries = retries;
+      if (remainingRetries-- > 0) {
+        remainingRetries -= 1;
+        return new Promise((resolve) => setTimeout(resolve, interval)).then(fn).catch(retry);
+      }
+      throw err;
+    });
+}
+
+// Method to invoke the create environment API for serverless.
+async function createEnvironment(
+  baseUrl: string,
```

Comment:
The function can read `baseUrl` from global variables as it does with `log`. If the function is moved to a separate `utility.ts` file or something it makes sense to pass `baseUrl`.

It would be good to transform input params into an object so it's visible what the function accepts

```
interface CreateEnvironmentParams {
  baseUrl: string;
  projectName: string;
  apiKey: string,
  ftrConfig: SecuritySolutionDescribeBlockFtrConfig;
}

async function createEnvironment({ baseUrl, projectName, apiKey, ftrConfig }: CreateEnvironmentParams): Promise<Environment> {
  ...
}
```

Reply:
Fixed with https://github.com/elastic/kibana/pull/169311/commits/591758012fc0136f4dd70641072443fd7da5908c

I did not actually change them to an object but left the three params there.